experiments/storm/src/main/scala/io/gearpump/experiments/storm/processor/StormProcessor.scala

```diff
@@ -47,17 +52,17 @@ private[storm] class StormProcessor(taskContext : TaskContext, conf: UserConfig)
   import StormUtil._
   import io.gearpump.experiments.storm.processor.StormProcessor._
 
-  private val topology = getTopology(conf)
-  private val pid = taskContext.taskId.processorId
-  private val boltId = conf.getString(GraphBuilder.COMPONENT_ID).getOrElse(
-    throw new RuntimeException(s"Storm bolt id not found for processor $pid"))
-  private val boltSpec = conf.getValue[Bolt](GraphBuilder.COMPONENT_SPEC).getOrElse(
-    throw new RuntimeException(s"Storm bolt spec not found for processor $pid"))
+  private val topology = conf.getValue[StormTopology](TOPOLOGY).get
+  private val boltId = conf.getString(GraphBuilder.COMPONENT_ID).get
+  private val boltSpec = conf.getValue[Bolt](GraphBuilder.COMPONENT_SPEC).get
   private val bolt = Utils.getSetComponentObject(boltSpec.get_bolt_object()).asInstanceOf[IBolt]
   private val stormConfig = getStormConfig(conf, boltSpec.get_common())
-  private val topologyContextBuilder = TopologyContextBuilder(topology, stormConfig,
-    multiLang = bolt.isInstanceOf[ShellBolt])
-  private val outputCollector = new StormOutputCollector(taskContext, pid, boltId)
+  private val taskToComponent = conf.getValue[Map[Integer, String]](TASK_TO_COMPONENT).get
+  private val componentToSortedTasks = getComponentToSortedTasks(taskToComponent)
+  private val stormTaskId = getStormTaskId(boltId, componentToSortedTasks, taskContext.taskId)
+  private val componentToStreamFields = getComponentToStreamFields(topology)
+  private val inputs: JMap[GlobalStreamId,  Grouping] = StormUtil.getInputs(boltId, topology)
+  private var outputCollector: StormOutputCollector = null
   private var scheduler: Cancellable = null
 
```

Comment:
The initialization list seems pretty long to me, do we need them all?

Whether it is a good idea to use TopologyContext to represent all? In that case, we can avoid passing so many values.